wire-tests/src/test/proto-kotlin/com/squareup/wire/protos/kotlin/OneOfMessage.kt

```diff
@@ -0,0 +1,79 @@
+// Code generated by Wire protocol buffer compiler, do not edit.
+// Source file: one_of.proto
+package com.squareup.wire.protos.kotlin
+
+import com.squareup.wire.FieldEncoding
+import com.squareup.wire.Message
+import com.squareup.wire.ProtoAdapter
+import com.squareup.wire.ProtoReader
+import com.squareup.wire.ProtoWriter
+import com.squareup.wire.TagHandler
+import kotlin.Deprecated
+import kotlin.DeprecationLevel
+import kotlin.Int
+import kotlin.String
+import kotlin.jvm.JvmField
+import okio.ByteString
+
+data class OneOfMessage(val choice: Choice? = null, val unknownFields: ByteString =
+        ByteString.EMPTY) : Message<OneOfMessage, OneOfMessage.Builder>(ADAPTER, unknownFields) {
+    @Deprecated(
+            message = "Shouldn't be used in Kotlin",
+            level = DeprecationLevel.HIDDEN
+    )
+    override fun newBuilder(): Builder = Builder(this.copy())
+
+    class Builder(private val message: OneOfMessage) : Message.Builder<OneOfMessage, Builder>() {
+        override fun build(): OneOfMessage = message
+    }
+
+    companion object {
+        @JvmField
+        val ADAPTER: ProtoAdapter<OneOfMessage> = object : ProtoAdapter<OneOfMessage>(
+            FieldEncoding.LENGTH_DELIMITED, 
+            OneOfMessage::class.java
+        ) {
+            override fun encodedSize(value: OneOfMessage): Int = 
+                when (value.choice) {
+                    is Choice.Foo -> ProtoAdapter.INT32.encodedSizeWithTag(1, value.choice.foo)
+                    is Choice.Bar -> ProtoAdapter.STRING.encodedSizeWithTag(3, value.choice.bar)
+                    is Choice.Baz -> ProtoAdapter.STRING.encodedSizeWithTag(4, value.choice.baz)
+                    else -> 0
+                } +
+                value.unknownFields.size
```

Comment:
nit: maybe that'd be easier to read if this line was first?
something like
```kotlin
value.unknownFields.size + when (value.choice) {
    is Choice.Foo -> ProtoAdapter.INT32.encodedSizeWithTag(1, value.choice.foo)
    is Choice.Bar -> ProtoAdapter.STRING.encodedSizeWithTag(3, value.choice.bar)
    is Choice.Baz -> ProtoAdapter.STRING.encodedSizeWithTag(4, value.choice.baz)
    else -> 0
}
```

Reply:
but on the other hand, we want unknowFields at the end

Reply:
Plus, this is just an edge case where the proto has a single field. Formatting multiple `oneof` fields in a meaningful way would be interesting.